modules/nf-core/merquryfk/merquryfk/tests/main.nf.test

```diff
@@ -0,0 +1,153 @@
+                    file(params.test_data['homo_sapiens']['illumina']['test_1_fastq_gz'], checkIfExists: true)
+                ]
+                """
+            }
+        }
+    }
+
+    test("test_merquryfk_merquryfk_png") {
+
+        when {
+            process {
+                """
+                assembly = [
+                    [ id:'test', single_end:true ], // meta map
+                    file(params.test_data['homo_sapiens']['genome']['genome_fasta'], checkIfExists: true)
+                ]
+
+                haplotigs = [
+                    [ id:'test', single_end:true ], []
+                ]
+
+                input[0] = FASTK_FASTK.out.hist
+                            | join( FASTK_FASTK.out.ktab )
+                            | join( Channel.value( assembly ) )
+                            | join( Channel.value( haplotigs ) )
+                """
+            }
+        }
+
+        then {
+            assertAll(
+                { assert process.success },
+                { assert snapshot(process.out).match() }
+            )
+        }
+    }
+
+    test("test_merquryfk_merquryfk_pdf") {
+
+        when {
+            process {
+                """
+                assembly = [
+                    [ id:'test', single_end:true ], // meta map
+                    file(params.test_data['homo_sapiens']['genome']['genome_fasta'], checkIfExists: true)
+                ]
+
+                haplotigs = [
+                    [ id:'test', single_end:true ], []
+                ]
+
+                input[0] = FASTK_FASTK.out.hist
+                            | join( FASTK_FASTK.out.ktab )
+                            | join( Channel.value( assembly ) )
+                            | join( Channel.value( haplotigs ) )
+                """
+            }
+        }
+
+        then {
+            assertAll(
+                { assert process.success },
+                { assert snapshot(process.out).match() }
+            )
+        }
+    }
+
+    test("single_minimal_stub") {
+
+        options '-stub'
+
+        when {
+            process {
+                """
+                assembly = [
+                    [ id:'test', single_end:true ], // meta map
+                    file(params.test_data['homo_sapiens']['genome']['genome_fasta'], checkIfExists: true)
+                ]
+
+                haplotigs = [
+                    [ id:'test', single_end:true ], []
+                ]
+
+                input[0] = FASTK_FASTK.out.hist
+                            | join( FASTK_FASTK.out.ktab )
+                            | join( Channel.value( assembly ) )
+                            | join( Channel.value( haplotigs ) )
+                """
+            }
+        }
+
+        then {
+            assertAll(
+                { assert process.success },
+                { assert snapshot(process.out).match() }
+            )
+        }
+    }
+
+    test("multiple_minimal_stub") {
```

Comment:
I am not sure: Why do you need two different stub tests? 🤔

Reply:
One runs with haplotigs and the other without it.